### ISSUE TITLE
Regression: thrown no-changed-when error for async operation with command

### ISSUE THREAD

**Issue by Red-GV** [ansible/ansible-lint#4649]
##### Summary

Original issue here: https://github.com/ansible/ansible-lint/issues/4638

There appears to be a regression from #1588 . 

`no-changed-when` is being thrown for an asynchronous operation when it should not. The [docs](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_async.html#run-tasks-concurrently-poll-0) state that `changed_when` should not be applied for an asynchronous task.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint 25.4.0 using ansible-core:2.18.3 ansible-compat:25.5.0 ruamel-yaml:0.18.12 ruamel-yaml-clib:0.2.12
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console (paste below)
- name: Do something with async
  ansible.builtin.command:
    cmd: echo 'hello world'
  async: 600
  poll: 0
  register: async_update_results
```

##### Desired Behavior

The `changed_when` error should not be firing.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

The `no-changed-when` error is appearing

```paste below
no-changed-when: Commands should not change things if nothing needs doing.
roles/awx/tasks/<name>.yaml: Task/Handler:
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by Red-GV** [comment]
Hello @ssbarnea , question. Why was the `bug` label removed? The linter still throws an error for this case despite the Ansible docs still advising against this behavior and the test cases to guard against this seemingly gone from the codebase (at least last I checked some months ago when I reported the issue).
